src/test/java/uk/ac/stfc/facilities/mailing/mailchimp/MailchimpListMemberRequestTest.java

```diff
@@ -1,52 +1,92 @@
-        assertThat(request.getStatus()).isNull();
+            assertThat(request.getEmail()).isEqualTo("test");
+            assertThat(request.getStatusIfNew()).isEqualTo("subscribed");
+            assertThat(request.getStatus()).isNull();
 
-    }
+        }
 
-    @Test
-    public void whichUnsubscribes() {
-        MailchimpListMemberRequest request = MailchimpListMemberRequest.whichUnsubscribes("test");
+        @Test
+        @DisplayName("for unsubscribing")
+        public void whichUnsubscribes() {
+            MailchimpListMemberRequest request = MailchimpListMemberRequest.whichUnsubscribes("test");
 
-        assertThat(request.getEmail()).isEqualTo("test");
-        assertThat(request.getStatusIfNew()).isEqualTo("unsubscribed");
-        assertThat(request.getStatus()).isNull();
+            assertThat(request.getEmail()).isEqualTo("test");
+            assertThat(request.getStatusIfNew()).isEqualTo("unsubscribed");
+            assertThat(request.getStatus()).isNull();
 
-    }
+        }
 
-    @Test
-    public void whichForceSubscribes() {
-        MailchimpListMemberRequest request = MailchimpListMemberRequest.whichForceSubscribes("test");
+        @Test
+        @DisplayName("for forcing subscribing")
+        public void whichForceSubscribes() {
+            MailchimpListMemberRequest request = MailchimpListMemberRequest.whichForceSubscribes("test");
 
-        assertThat(request.getEmail()).isEqualTo("test");
-        assertThat(request.getStatusIfNew()).isEqualTo("subscribed");
-        assertThat(request.getStatus()).isEqualTo("subscribed");
+            assertThat(request.getEmail()).isEqualTo("test");
+            assertThat(request.getStatusIfNew()).isEqualTo("subscribed");
+            assertThat(request.getStatus()).isEqualTo("subscribed");
 
-    }
+        }
+
+        @Test
+        @DisplayName("for forcing unsubscribing")
+        public void whichForceUnsubscribes() {
+            MailchimpListMemberRequest request = MailchimpListMemberRequest.whichForceUnsubscribes("test");
 
-    @Test
-    public void whichForceUnsubscribes() {
-        MailchimpListMemberRequest request = MailchimpListMemberRequest.whichForceUnsubscribes("test");
+            assertThat(request.getEmail()).isEqualTo("test");
+            assertThat(request.getStatusIfNew()).isEqualTo("unsubscribed");
+            assertThat(request.getStatus()).isEqualTo("unsubscribed");
 
-        assertThat(request.getEmail()).isEqualTo("test");
-        assertThat(request.getStatusIfNew()).isEqualTo("unsubscribed");
-        assertThat(request.getStatus()).isEqualTo("unsubscribed");
+        }
 
     }
+
+    @DisplayName("when called with a null email arguments should throw an IllegalArgumentException")
+    public class Failure {
+
+        @Test
+        @DisplayName("for subscribing")
+        public void whichSubscribes() {
+            assertThrows(IllegalArgumentException.class, () -> MailchimpListMemberRequest.whichSubscribes("test"));
```

Comment:
Should these be passing null instead of "test"? These look like they won't throw an exception, but I might be misunderstanding.

Reply:
You're entirely correct and point out the problem with me not doing TDD properly. 😆